src/CoreBundle/DataModel/BackendTreeNodeDataModel.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\DataModel;
+
+use JsonSerializable;
+
+class BackendTreeNodeDataModel implements JsonSerializable
+{
+    /**
+     * @var string
+     */
+    private $id = '';
+
+    /**
+     * @var string
+     */
+    private $name = '';
+
+    /**
+     * @var int
+     */
+    private $cmsIdent = '';
+
+    /**
+     * @var BackendTreeNodeDataModel[]
+     */
+    private $children = [];
+
+    /**
+     * @var bool
+     */
+    private $childrenAjaxLoad = false;
+
+    /**
+     * @var string
+     */
+    private $type = '';
+
+    /**
+     * @var bool
+     */
+    private $selected = false;
+
+    /**
+     * @var bool
+     */
+    private $disabled = false;
+
+    /**
+     * @var bool
+     */
+    private $opened = false;
+
+    /**
+     * Key = HTML attribute name.
+     *
+     * @var array
+     */
+    private $listAttributes = [];
+
+    /**
+     * Key = HTML attribute name.
+     *
+     * @var array
+     */
+    private $linkAttributes = [];
+
+    /**
+     * @var array
+     */
+    private $linkHtmlClasses = [];
+
+    /**
+     * @var array
+     */
+    private $listHtmlClasses = [];
+
+    /**
+     * @var string
+     */
+    private $connectedPageId;
+
+    public function __construct($id, $name, $cmsIdent, $connectedPageId)
```

Comment:
__construct(int $id, string $name, string $cmsIdent, string $connectedPageId)

Reply:
ok, but $id is a string and $cmsIdent an int

Reply:
of course... sorry. this way around.